### PR TITLE
Cl 10 GB language search bar changes

### PR DESCRIPTION
**Pull request by gunjanbajaj1**
@xDAnkit I have added the changes for the Language search Bra.

Below are the details:

1. The search bar is added in the language selection menu.
2. The selected Language is now persistent even the user reloads the page.
3. The recently selected languages are also present within the language selection menu.

### CODE REVIEW

File: client/src/EditorTab.jsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ export default function EditorTab() {
         value={codeSave ? codeSave.code : ""}
         onChange={preTextRemove}
         options={{
-          readOnly: !hasUser,
+          //readOnly: !hasUser,
```
Comment (xDAnkit) [inline client/src/EditorTab.jsx]: @gunjanbajaj1 - Please uncomment this line

File: client/src/NavBar.jsx [hunk 2]
```diff
@@ -1,111 +1,121 @@
...
+
+    for (let key in supportedLangList) {
+      const currentLang = supportedLangList[key];
+
+      if (
+        search &&
+        currentLang.toLowerCase().search(search.toLowerCase()) !== -1
+      ) {
+        results.push({ id: currentLang, value: key });
+      } else if (!search) {
```
Comment (xDAnkit) [inline client/src/NavBar.jsx]: @gunjanbajaj1 - Convert this into a guard-block code

File: client/src/NavBar.jsx [hunk 3]
```diff
@@ -124,27 +134,30 @@ export default function NavBar(props) {
               Home
             </Link>
             {Location.pathname.startsWith("/") ? (
```
Comment (xDAnkit) [inline client/src/NavBar.jsx]: @gunjanbajaj1 - Do we really need this?

**Review by xDAnkit — CHANGES_REQUESTED** [review]
@gunjanbajaj1 - Please check comments

File: client/src/EditorTab.jsx (hunk 1 above)
Comment (gunjanbajaj1) [reply client/src/EditorTab.jsx]: @xDAnkit Done

File: client/src/NavBar.jsx (hunk 2 above)
Comment (gunjanbajaj1) [reply client/src/NavBar.jsx]: @xDAnkit Fixed

File: client/src/NavBar.jsx (hunk 3 above)
Comment (gunjanbajaj1) [reply client/src/NavBar.jsx]: @xDAnkit Done

File: client/src/LanguageSelector.jsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    const selectedKey = "PlainText";
+    const lastSelectedLanguage = getLanguage(LAST_SELECTED_LANGUAGE);
+    console.log(lastSelectedLanguage);
+    setSelectLang(lastSelectedLanguage ? lastSelectedLanguage : selectedKey);
+  }, []);
+
+  function onSelectLang(value) {
+    setSelectLang(value);
+    setLanguage(LAST_SELECTED_LANGUAGE, value);
+    const existingLanguage = JSON.parse(getLanguage(LANGUAGE_HISTORY)) || [];
```
Comment (xDAnkit) [inline client/src/LanguageSelector.jsx]: @gunjanbajaj1 - Move this parse logic to the file

File: client/src/LanguageSelector.jsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    existingLanguage.includes(value)
+      ? existingLanguage.splice(existingLanguage.indexOf(value), 1)
+      : existingLanguage;
+    existingLanguage.push(value);
+    setLanguage(LANGUAGE_HISTORY, JSON.stringify(existingLanguage));
```
Comment (xDAnkit) [inline client/src/LanguageSelector.jsx]: @gunjanbajaj1 - Move this stringify logic to the file

File: client/src/LanguageSelector.jsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+            value={search}
+          />
+          <ScrollableMenu>{_renderLanguages()}</ScrollableMenu>
+        </Dropdown.Menu>
+      </Dropdown>
+    </>
+  );
+}
+
+const ScrollableMenu = forwardRef(
```
Comment (xDAnkit) [inline client/src/LanguageSelector.jsx]: @gunjanbajaj1 - Move this inside `components/templates`

**Review by xDAnkit — CHANGES_REQUESTED** [review]
@gunjanbajaj1 - Pls check comments

**Review by xDAnkit — APPROVED** [review]
Looks good to me

**Pull request merged**